.github/branch-protection.yml

```
@@ -0,0 +1,256 @@
+  allow_auto_merge: true
+  delete_branch_on_merge: true
+  allow_update_branch: true
+
+  # Default branch
+  default_branch: main
+
+# Code Owners Configuration
+codeowners:
+  # Global owners
+  "*": ["@admin", "@lead-developer"]
+  
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)